src/app/features/tech-record/components/tech-record-amend-vrm/tech-record-amend-vrm.component.ts

```diff
@@ -63,80 +103,69 @@ export class AmendVrmComponent implements OnDestroy, OnInit {
     this.actions$.pipe(ofType(amendVrmSuccess), takeUntil(this.destroy$)).subscribe(({ vehicleTechRecord }) => {
       this.router.navigate(['/tech-records', `${vehicleTechRecord.systemNumber}`, `${vehicleTechRecord.createdTimestamp}`]);
     });
+
+    this.cherishedTransferForm.controls['previousVrm'].setValue(this.techRecord?.primaryVrm ?? '');
+    this.cherishedTransferForm.controls['previousVrm'].disable();
+    this.cherishedTransferForm.controls['thirdMark'].setAsyncValidators(
+      this.technicalRecordService.validateVrmDoesNotExist(this.techRecord?.primaryVrm ?? '')
+    );
+    this.correctingAnErrorForm.controls['newVrm'].setAsyncValidators(
+      this.technicalRecordService.validateVrmDoesNotExist(this.techRecord?.primaryVrm ?? '')
+    );
   }
 
   ngOnDestroy(): void {
     this.destroy$.next();
     this.destroy$.complete();
   }
 
-  get reasons(): Array<FormNodeOption<boolean>> {
-    return [
-      { label: 'Cherished transfer', value: true, hint: 'Current VRM will be archived' },
-      { label: 'Correcting an error', value: false, hint: 'Current VRM will not be archived' }
-    ];
-  }
-
-  get width(): FormNodeWidth {
-    return FormNodeWidth.L;
-  }
-
   get vehicleType(): VehicleTypes | undefined {
     return this.techRecord ? this.technicalRecordService.getVehicleTypeWithSmallTrl(this.techRecord) : undefined;
   }
 
   navigateBack() {
     this.globalErrorService.clearErrors();
-    this.router.navigate(['..'], { relativeTo: this.route });
+    this.router.navigate(['../../'], { relativeTo: this.route });
+  }
+
+  handleFormChange() {
+    if (this.isCherishedTransfer) {
+      this.cherishedTransferForm.get('currentVrm')?.updateValueAndValidity();
+      return;
+    }
   }
 
   handleSubmit(): void {
-    if (!this.isFormValid()) return;
-    this.globalErrorService.errors$
-      .pipe(
-        take(1),
-        filter(errors => !errors.length),
-        switchMap(() => this.technicalRecordService.isUnique(this.form.value.newVrm, SEARCH_TYPES.VRM)),
-        take(1),
-        catchError(error => (error.status == 404 ? of(true) : throwError(() => new Error('Error'))))
-      )
-      .subscribe({
-        next: res => {
-          if (!res) return this.globalErrorService.addError({ error: 'VRM already exists', anchorLink: 'newVrm' });
-
-          this.store.dispatch(
-            amendVrm({
-              newVrm: this.form.value.newVrm,
-              cherishedTransfer: this.form.value.isCherishedTransfer,
-              systemNumber: (this.techRecord as TechRecordType<'get'>)?.systemNumber!,
-              createdTimestamp: (this.techRecord as TechRecordType<'get'>)?.createdTimestamp!
-            })
-          );
-        },
-        error: e => this.globalErrorService.addError({ error: 'Internal Server Error', anchorLink: 'newVrm' })
-      });
+    if (!this.isFormValid()) {
+      return;
+    }
+
+    this.store.dispatch(
```

Comment:
I've refactored this dispatch so it's using ternaries for the newVrrm and thirdMark, I'm not too sure how readable this is though.

Reply:
It looks okay to me